src/overlays/actors/ovl_En_Fish/z_en_fish.c

```diff
@@ -42,87 +81,875 @@ const ActorInit En_Fish_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091E880.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091E9A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091EAF0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091ECF4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091ED70.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091EF30.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091EFE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091F344.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091F3BC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091F5A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091F830.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/EnFish_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/func_8091F994.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Fish/EnFish_Draw.s")
+f32 func_8091D630(Vec3f* arg0, Vec3f* arg1) {
+    return SQ(arg0->x - arg1->x) + SQ(arg0->z - arg1->z);
+}
+
+void func_8091D660(EnFish* this) {
+    Animation_Change(&this->skelAnime, &D_0402F0EC, 1.0f, 0.0f, Animation_GetLastFrame(&D_0402F0EC), 1, 2.0f);
+}
+
+void func_8091D6C4(EnFish* this) {
+    Animation_Change(&this->skelAnime, &D_0402E65C, 1.0f, 0.0f, Animation_GetLastFrame(&D_0402E65C), 1, 2.0f);
+}
+
+void func_8091D728(EnFish* this) {
+    f32 phi_f0;
+
+    this->actor.scale.x = this->unk_250 * this->unk_25C;
+    this->actor.scale.y = this->unk_254 * this->unk_25C;
+    this->actor.scale.z = this->unk_258 * this->unk_25C;
+
+    if (this->actor.scale.z > 0.00001f) {
+        phi_f0 = 1.0f / this->actor.scale.z;
+    } else {
+        phi_f0 = 0.0f;
+    }
+    this->unk_264 = 16.0f * phi_f0;
+    this->unk_266 = 12.0f * phi_f0;
+}
+
+void func_8091D7C4(EnFish* this) {
+    s32 pad;
+
+    Math_ScaledStepToS(&this->actor.world.rot.x, this->unk_268, this->unk_26E);
+    Math_ScaledStepToS(&this->actor.world.rot.y, this->unk_26A, this->unk_270);
+    if (this->actor.world.rot.z != this->unk_26C) {
+        Math_ScaledStepToS(&this->actor.world.rot.z, this->unk_26C, this->unk_272);
+    }
+    this->actor.shape.rot = this->actor.world.rot;
+}
+
+void func_8091D840(Actor* thisx, GlobalContext* globalCtx, s32 arg2, f32 arg3) {
+    EnFish* this = THIS;
+    s32 i;
+
+    for (i = 0; i < arg2; i++) {
+        EffectSsBubble_Spawn(globalCtx, &this->actor.world.pos, 0.0f, arg3, arg3, (Rand_ZeroOne() * 0.09f) + 0.06f);
+    }
+}
+
+void func_8091D904(EnFish* this) {
+    this->unk_242 = 400;
+    this->unk_25C = this->unk_260 * 0.1f;
+    func_8091D728(this);
+    this->actor.draw = NULL;
+}
+
+Actor* func_8091D944(EnFish* this, GlobalContext* globalCtx) {
+    f32 dist;
+    Actor* retActor = NULL;
+    f32 minDist = FLT_MAX;
```

Comment:
```suggestion
    f32 distSq;
    Actor* retActor = NULL;
    f32 minDistSq = FLT_MAX;
```
These are square of distances, as opposed to distances